pkg/memtier/counters_heatmap_test.go

```diff
@@ -48,59 +48,48 @@ func TestUpdateCountersBoundaryCheck(t *testing.T) {
 	t.Logf(hm.Dump())
 
 	// Boundary value check: nil/non-nil
-	if hm.HeatRangeAt(4040, 0) != nil {
-		t.Errorf("nil expected when requesting heat for non-existing pid")
-	}
-	if hm.HeatRangeAt(2000, 99*PS) != nil {
-		t.Errorf("nil expected when requesting heat for address before any range")
-	}
-	if hm.HeatRangeAt(2000, 100*PS) == nil {
-		t.Errorf("non-nil expected when requesting heat for the start address in the first range")
-	}
-	if hm.HeatRangeAt(2000, 150*PS) == nil {
-		t.Errorf("non-nil expected when requesting heat for the start address in the second range")
-	}
-	if hm.HeatRangeAt(2000, 199*PS) == nil {
-		t.Errorf("non-nil expected when requesting heat for the last address in the second range")
-	}
-	if hm.HeatRangeAt(2000, 200*PS) == nil {
-		t.Errorf("non-nil expected when requesting heat for the first address in the third range")
-	}
-	if hm.HeatRangeAt(2000, 249*PS) == nil {
-		t.Errorf("non-nil expected when requesting heat for the last address in the third range")
-	}
-	if hm.HeatRangeAt(2000, 250*PS) != nil {
-		t.Errorf("nil expected when requesting heat for the first address in the hole")
-	}
-	if hm.HeatRangeAt(2000, 499*PS) != nil {
-		t.Errorf("nil expected when requesting heat for the last address in the hole")
-	}
-	if hm.HeatRangeAt(2000, 500*PS) == nil {
-		t.Errorf("non-nil expected when requesting heat for the first address after the hole")
-	}
-	if hm.HeatRangeAt(2000, 599*PS) == nil {
-		t.Errorf("non-nil expected when requesting heat for the last address after the hole")
+	boundryTestScenarios := []struct {
+		pid     int
+		addr    uint64
+		message string
+		result  bool
```

Comment:
Let's rename "result" to "expectNil" to describe the meaning of the boolean value. And let's use the same order of fields in boundaryTcases and heatTcases: inputs, expected outcome, and error message.

I'm also suggesting to using name (postifx) "tcases" instead of "testscenario", as "tcases" is currently used in all other unit tests for list of these input/expect structs, too.